src/components/PeopleTable.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from 'classnames';
+import { Person } from '../types';
+import { useEffect, useState } from 'react';
+import { PersonLink } from './PersonLink';
+import { useParams } from 'react-router-dom';
+
+type PeopleTableProps = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({ people }) => {
+  const { slug } = useParams();
+  const [selectPerson, setSelectPerson] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (slug) {
+      setSelectPerson(slug);
+    }
+  }, [slug]);
+
+  const findPersonByName = (name: string | undefined | null): Person | null =>
+    name
+      ? people.find(p => p.name.toLowerCase() === name.toLowerCase()) || null
+      : null;
```

Comment:
The function `findPersonByName` is called multiple times with the same argument within the JSX. It would be more efficient to call it once and store the result in a variable, then use that variable in the JSX. This will prevent unnecessary repeated calculations.